src/vsr/replica.zig

```diff
@@ -2178,24 +2179,18 @@ pub fn ReplicaType(
                 return;
             }
 
+            if (self.grid.canceling) |_| {
```

Comment:
Hm, shouldnt' we look at `self.syncing` here instead and assert grid.cancelling? `self.syncing` matches the current abstraction (replica) better.

Reply:
I was thinking we want to be liberal in terms of "who can respond to `request_blocks`", since it is technically possible for the majority of the cluster (2/3) to state sync simultaneously.

See: https://github.com/tigerbeetle/tigerbeetle/blob/2e7f7b6f2ce5c2a2932a6fa9130549be8bdcd15d/src/vsr/replica_test.zig#L719-L726 and https://github.com/tigerbeetle/tigerbeetle/blob/2e7f7b6f2ce5c2a2932a6fa9130549be8bdcd15d/src/vsr/replica.zig#L4280-L4306